app/controllers/hub/clients_controller.rb

```diff
@@ -56,8 +56,15 @@ def update
     end
 
     def response_needed
-      @client.clear_response_needed if params.fetch(:client, {})[:action] == "clear"
-      @client.set_response_needed! if params.fetch(:client, {})[:action] == "set"
+      case params.fetch(:client, {})[:action]
```

Comment:
whoa what is `params.fetch(:client, {})`? specifically why the empty hash what does it do

Reply:
Good question! that was existing code that I did not reexamine.  It appears to try to grab client params, and defaults to an empty hash if not available. However, this should probably be utilizing the standard secure parameters pattern, where we have a `response_needed` method that calls `params.require(:client).permit(:action)`